base/compiler/tfuncs.jl

```diff
@@ -331,7 +331,7 @@ end
 add_tfunc(Core.ifelse, 3, 3, ifelse_tfunc, 1)
 
 @nospecs function ifelse_nothrow(𝕃::AbstractLattice, cond, x, y)
-    ⊑ = Core.Compiler.:⊑(𝕃)
+    ⊑ = partialorder(𝕃)
```

Comment:
FWIW, the `Compiler.:⊑` is okay, just not Core to make it global. Or this very annoying syntax with let global to rebind it:
```
julia> using Core.Compiler: ⊑

julia> f(𝕃) = (⊑ = let; global ⊑; ⊑(𝕃); end)
f (generic function with 1 method)

julia> f(𝕃)
```

Reply:
I think whether `⊑ = partialorder(𝕃)` is clearer than `⊑ = Compiler.:⊑(𝕃)`. In the same vein, I'm also thinking about replacing `tmerge` with `⊔ = join(𝕃)`. What do you think?

Reply:
That seems fine too. Though maybe a bit confusing that it is not an optimal join on the 𝕃 (that would be `Union`), but rather on the complexity co-lattice

Reply:
Got it. Then, we might consider refactoring to switch to the 2-arg form using `partialorder(𝕃)` or `widening(𝕃)` (or `weakjoin(𝕃)`?). This could turn into a significant refactor, so it might be best to undertake it after finishing the upcoming expected refactors.
Anyway, I would like to move this PR forward for now. @Keno please let me know if you have a preference for the current 3-arg form.